plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -304,6 +318,108 @@ private static String quotedName(String name)
         return '"' + name.replace("\"", "\"\"") + '"';
     }
 
+    public static boolean canEnforceColumnConstraintInAllSpecs(TypeOperators typeOperators, Table table, IcebergColumnHandle columnHandle, Domain domain)
+    {
+        return table.specs().values().stream()
+                .allMatch(spec -> canEnforceConstraintWithinPartitioningSpec(typeOperators, spec, columnHandle, domain));
+    }
+
+    private static boolean canEnforceConstraintWithinPartitioningSpec(TypeOperators typeOperators, PartitionSpec spec, IcebergColumnHandle column, Domain domain)
+    {
+        for (PartitionField field : spec.fields()) {
+            if (field.transform().toString().equals("void")) {
+                // Useless for filtering.
+                continue;
+            }
+            if (field.sourceId() != column.getId()) {
+                // Not related to each other.
+                continue;
+            }
+            if (field.transform().isIdentity()) {
+                // A predicate on an identity partitioning column can always be enforced.
+                return true;
+            }
+
+            ColumnTransform transform = PartitionTransforms.getColumnTransform(field, column.getType());
+            if (transform.isMayReturnNullForNonNull()) {
+                // Partitioning transform must return NULL for NULL input.
+                // Below we assume it never returns NULL for non-NULL input,
+                // so NULL values and non-NULL values are always segregated.
+                // In practice, this condition matches the void transform only,
+                // which isn't useful for filtering anyway.
+                continue;
+            }
+            ValueSet valueSet = domain.getValues();
+
+            boolean canEnforce = valueSet.getValuesProcessor().transform(
+                    ranges -> {
+                        MethodHandle targetTypeEqualOperator = typeOperators.getEqualOperator(
```

Comment:
Please add a comment explaining the reasoning behind this assignment.

The parameter seems a bit of magic code without a proper explanation:

```
InvocationConvention.simpleConvention(FAIL_ON_NULL, NEVER_NULL, NEVER_NULL)
```